test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/RemoteAttributeTest.cs

```diff
@@ -155,6 +159,70 @@ public void Constructor_WithActionControllerAreaName_UpdatesAreaRouteData(string
             Assert.Null(attribute.RouteName);
         }
 
+        [Fact]
+        public void ErrorMessageProperties_HaveExpectedDefaultValues()
+        {
+            // Arrange & Act
+            var attribute = new RemoteAttribute("Action", "Controller");
+
+            // Assert
+            Assert.Null(attribute.ErrorMessage);
+            Assert.Null(attribute.ErrorMessageResourceName);
+            Assert.Null(attribute.ErrorMessageResourceType);
+        }
+
+        [Fact]
+        [ReplaceCulture]
+        public void FormatErrorMessage_ReturnsDefaultErrorMessage()
+        {
+            // Arrange
+            // See ViewFeatures.Resources.RemoteAttribute_RemoteValidationFailed.
+            var expected = "'Property1' is invalid.";
+            var attribute = new RemoteAttribute("Action", "Controller");
+
+            // Act
+            var message = attribute.FormatErrorMessage("Property1");
+
+            // Assert
+            Assert.Equal(expected, message);
+        }
+
+        [Fact]
+        public void FormatErrorMessage_UsesOverriddenErrorMessage()
+        {
+            // Arrange
+            var expected = "Error about 'Property1' from override.";
+            var attribute = new RemoteAttribute("Action", "Controller")
+            {
+                ErrorMessage = "Error about '{0}' from override.",
+            };
+
+            // Act
+            var message = attribute.FormatErrorMessage("Property1");
+
+            // Assert
+            Assert.Equal(expected, message);
+        }
+
+        [Fact]
+        [ReplaceCulture]
+        public void FormatErrorMessage_UsesErrorMessageFromResource()
```

Comment:
This test and any other that accessed the `protected string ErrorMessageString` from ValidationAttribute would have failed w/o `RemoteAttribute` constructor change.